test/e2e/util_test.go

```diff
@@ -137,7 +137,11 @@ func buildCurlPod(name, namespace, image, host, address string, extraArgs ...str
 		"--retry", "300", "--retry-delay", "1", "--max-time", "2",
 	}
 	curlArgs = append(curlArgs, extraArgs...)
-	curlArgs = append(curlArgs, "http://"+host)
+	if strings.HasPrefix(host, "http://") || strings.HasPrefix(host, "https://") {
```

Comment:
I think the only reason you need to do this is because you're borrowing the canary route.  Please don't borrow the canary route, as mentioned previously.

Reply:
I think it would be good for `buildCurlPod` to allow https, but with the switch to an echo pod, it doesn't need to be done here, so I've removed this change.